api/src/backend/api/base_views.py

```diff
@@ -29,6 +31,17 @@ class BaseViewSet(ModelViewSet):
     ordering_fields = "__all__"
     ordering = ["id"]
 
+    def initial(self, request, *args, **kwargs):
+        """
+        Sets required_permissions before permissions are checked.
+        """
+        self.get_required_permissions()
+        super().initial(request, *args, **kwargs)
+
+    def get_required_permissions(self):
+        """This is an abstract method that must be implemented by subclasses."""
+        NotImplemented
```

Comment:
There are a few viewsets that do not implement this method. How is it working?

Reply:
They use their required_permissions parameter, so the function get_required_permissions does not return anything, maybe the interesting thing would be to change the name of this function to set_required_permissions.